src/child/child.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { ChildCreate, ChildUpdate } from 'src/interfaces/child.interface';
+import { Child } from 'src/schemas/Child.schema';
+import { Class } from 'src/schemas/Class.schema';
+import { Parent } from 'src/schemas/Parent.schema';
+
+@Injectable()
+export class ChildService {
+  constructor(
+    @InjectModel(Child.name) private readonly childModel: Model<Child>,
+    @InjectModel(Class.name) private readonly classModel: Model<Class>,
+    @InjectModel(Parent.name) private readonly parentModel: Model<Parent>,
+  ) {}
+
+  async create(childCreate: ChildCreate) {
+    const parent = await this.parentModel.findById(childCreate.parentId);
+    if (!parent) {
+      throw new NotFoundException('Parent not found');
+    }
+    const clazz = await this.classModel.findById(childCreate.classId);
+    if (!clazz) {
+      throw new NotFoundException('Class not found');
+    }
+    return await this.childModel.create({
+      ...childCreate,
+      parent: parent._id,
+      class: clazz._id,
+    });
```

Comment:
wrap this in try catch, because if something goes wrong in mongoose methods we would not catch that and internal server error will be thrown

Reply:
do it for the rest of methods